src/reducers/userReducer.js

```diff
@@ -6,17 +6,35 @@ let initialState = {
 };
 
 function userReducer(state = initialState, action) {
+  let newUsers = [];
+
   switch (action.type) {
     case ActionTypes.CREATE_USER:
       return Object.assign({}, state, { currUser: action.user });
     case ActionTypes.ADD_USERS:
       return Object.assign({}, state, { users: action.users });
+    case ActionTypes.ADD_VALUE:
+      newUsers = JSON.parse(JSON.stringify(state.users));
+      newUsers.forEach(elem => {
+        action.userValue.forEach( item => {
+          if (elem.id === item.id) {
+            elem.value = item.value
+          }
+        })
+      })
+      return Object.assign({}, state, { users: newUsers });
+    case ActionTypes.DROP_VALUES:
+      newUsers = JSON.parse(JSON.stringify(state.users));
+      newUsers.forEach(elem => {
+        elem.value = null;
+      })
+      return Object.assign({}, state, { users: newUsers });
     case ActionTypes.REMOVE_USER:
       let userIdx = state.users.map(elem => elem.id).indexOf(action.userId);
-      let newUsers = JSON.parse(JSON.stringify(state.users));
+      newUsers = JSON.parse(JSON.stringify(state.users));
```

Comment:
i think this way should be fine unless state.users has methods
also you can use Object assign i guess